sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionalExpressions.scala

```diff
@@ -182,11 +183,16 @@ case class CaseWhen(branches: Seq[(Expression, Expression)], elseValue: Option[E
 
     generatedCode += "}\n" * cases.size
 
-    s"""
+    if (generatedCode.size > 64 * 1000) {
```

Comment:
I guess you mean 10K? That makes sense. Should we modify the corresponding check in CodegenContext.splitExpressions?